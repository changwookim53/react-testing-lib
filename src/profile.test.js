import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";
describe("<Profile />", () => {
    it("matches snapshot", () => {
        const utils = render(<Profile username="changwookim53" name="김창우" />);
        expect(utils.container).toMatchSnapshot();
    });
    it("shows the props correctly", () => {
        const utils = render(<Profile username="changwookim53" name="김창우" />);
        utils.getByText("changwookim53");
        utils.getByText("(김창우)");
        // utils.getByText("changwookim53 (김창우)");
    });
});

// 2개의 테스트를 it으로 표현하여 넣었다.
// react가 동작되는 것이 머릿속으로 그려져야 한다./