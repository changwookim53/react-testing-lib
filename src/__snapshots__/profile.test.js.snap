// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      changwookim53
    </b>
    <span>
      (
      김창우
      )
    </span>
  </div>
</div>
`;
